`ioctl ioid register` crashing on mainnet

```sh
 ioctl config set endpoint api.iotex.one:443
 ioctl ioid register myproject  


panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/iotexproject/iotex-core/ioctl/cmd/ioid.register({0xc00068b8d0?, 0x0?, 0x0?})
	/home/back/Source/GitHub/iotex-core/ioctl/cmd/ioid/projectregister.go:96 +0x2e5
github.com/iotexproject/iotex-core/ioctl/cmd/ioid.init.func5(0xc00014ca00?, {0xc00068b8d0?, 0x4?, 0x1243986?})
	/home/back/Source/GitHub/iotex-core/ioctl/cmd/ioid/projectregister.go:43 +0x25
github.com/spf13/cobra.(*Command).execute(0x1ebe1a0, {0xc00068b8b0, 0x1, 0x1})
	/home/back/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x6a2
github.com/spf13/cobra.(*Command).ExecuteC(0xc00054e588)
	/home/back/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x38d
github.com/spf13/cobra.(*Command).Execute(0xc000006380?)
	/home/back/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918 +0x13
main.main()
	/home/back/Source/GitHub/iotex-core/tools/ioctl/ioctl.go:16 +0x18
```

--- a/ioctl/cmd/account/accountcreateadd.go
+++ b/ioctl/cmd/account/accountcreateadd.go
@@ -15,6 +15,7 @@ import (
 	"github.com/iotexproject/iotex-core/v2/ioctl/config"
 	"github.com/iotexproject/iotex-core/v2/ioctl/output"
 	"github.com/iotexproject/iotex-core/v2/ioctl/validator"
+	"github.com/iotexproject/iotex-core/v2/pkg/util/addrutil"
 )
 
 // Multi-language support
@@ -81,7 +82,13 @@ func accountCreateAdd(args []string) error {
 		return output.NewError(output.WriteFileError,
 			fmt.Sprintf("failed to write to config file %s", config.DefaultConfigFile), err)
 	}
+	ethAddr, err := addrutil.IoAddrToEvmAddr(addr)
+	if err != nil {
+		return output.NewError(output.AddressError, "failed to convert to ETH address", err)
+	}
 	output.PrintResult(fmt.Sprintf("New account \"%s\" is created.\n"+
-		"Please Keep your password, or you will lose your private key.", alias))
+		"IO address:  %s\n"+
+		"ETH address: %s\n"+
+		"Please Keep your password, or you will lose your private key.", alias, addr, ethAddr.String()))
 	return nil
 }

--- a/ioctl/cmd/bc/bcinfo.go
+++ b/ioctl/cmd/bc/bcinfo.go
@@ -56,7 +56,9 @@ func (m *infoMessage) String() string {
 func bcInfo() error {
 	chainMeta, err := GetChainMeta()
 	if err != nil {
-		return output.NewError(0, "failed to get chain meta", err)
+		return output.NewError(0, "failed to get chain meta; if you are querying a local node, "+
+			"it may still be syncing or its indexer may not be up to date yet — wait until the node "+
+			"has caught up, or set --endpoint to a fully synced node", err)
 	}
 	message := infoMessage{Node: config.ReadConfig.Endpoint, Info: chainMeta}
 	fmt.Println(message.String())

--- a/ioctl/cmd/ioid/projectregister.go
+++ b/ioctl/cmd/ioid/projectregister.go
@@ -93,7 +93,13 @@ func register(args []string) error {
 		return output.NewError(output.UpdateError, "failed to register project", err)
 	}
 
-	projectId := new(big.Int).SetBytes(receipt.ReceiptInfo.Receipt.Logs[0].Topics[3])
+	logs := receipt.ReceiptInfo.Receipt.Logs
+	if len(logs) == 0 || len(logs[0].Topics) < 4 {
+		return output.NewError(output.APIError,
+			fmt.Sprintf("project registration tx %s was mined but no project-id log was found; "+
+				"the registration may have reverted, please check the receipt on the explorer", tx), nil)
+	}
+	projectId := new(big.Int).SetBytes(logs[0].Topics[3])
 	fmt.Printf("Registered ioID project id is %s\n", projectId.String())
 
 	return nil

--- a/ioctl/cmd/update/update.go
+++ b/ioctl/cmd/update/update.go
@@ -7,6 +7,7 @@ package update
 
 import (
 	"fmt"
+	"os"
 	"os/exec"
 
 	"github.com/spf13/cobra"
@@ -66,6 +67,10 @@ func update() error {
 
 	}
 	cmd := exec.Command("bash", "-c", cmdString)
+	// surface the installer's stdout/stderr so failures are diagnosable instead
+	// of being swallowed behind a generic "failed to update ioctl" message.
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
 	output.PrintResult(fmt.Sprintf("Downloading the latest %s version ...\n", _versionType))
 	err := cmd.Run()
 	if err != nil {

--- a/ioctl/cmd/version/version.go
+++ b/ioctl/cmd/version/version.go
@@ -20,8 +20,8 @@ import (
 // Multi-language support
 var (
 	_versionCmdShorts = map[config.Language]string{
-		config.English: "Print the version of ioctl and node",
-		config.Chinese: "打印ioctl和节点的版本",
+		config.English: "Print the version and build info of ioctl",
+		config.Chinese: "打印ioctl的版本与构建信息",
 	}
 	_flagEndpointUsage = map[config.Language]string{
 		config.English: "set endpoint for once",

